buildSrc/src/main/groovy/org/elasticsearch/gradle/VersionCollection.groovy

```diff
@@ -25,159 +25,293 @@ import java.util.regex.Matcher
 
 /**
  * The collection of version constants declared in Version.java, for use in BWC testing.
+ *
+ *  if major+1 released: released artifacts from $version down to major-1.highestMinor.highestPatch, none of these should be snapshots, period.
```

Comment:
Ahh yea i can fix this up. Its because major is *current version* major, but it does not really say that well so its confusing ++

Reply:
ahh, ok so this should be as is, since its saying if your next major version is released (IE, im on 5.x currently), then go from $version (some 5.x) down to 2.whatever_the_highest_minor_is, which is the wire compat guarantee. It is slightly confusing tho, and i welcome a rewording help, i just cant think of a better way currently.